prototype/distribution/launcher/src/main/java/org/eclipse/sensinact/gateway/launcher/ConfigurationManager.java

```diff
@@ -196,4 +223,69 @@ private void createConfig(String pid, Hashtable<String, Object> value) throws IO
         }
         cfg.update(value);
     }
+
+    private Hashtable<String, Object> mergeConfigs(final Map<String, Object> currentValues,
+            final Map<String, Object> defaultValues) {
+        final Hashtable<String, Object> newValues = new Hashtable<>(defaultValues);
+        newValues.putAll(currentValues);
+        return newValues;
+    }
+
+    public void updateConfigurations(final Map<String, Hashtable<String, Object>> newConfigurations,
+            Collection<String> removedConfigurations) throws IOException {
+
+        // Load current status
+        final ConfigurationResource configResource = loadConfigFile();
+        final Map<String, Hashtable<String, Object>> allConfigurations = configResource.getConfigurations();
+
+        // Ensure that the configuration resource has the mandatory properties
+        final Map<String, Object> configProperties = configResource.getProperties();
+        if(configProperties.isEmpty()) {
+            // Generate the mandatory properties
+            configResource.getProperties().put(":configurator:resource-version", 1);
+            // TODO: use the bundle context to get those?
+            configResource.getProperties().put(":configurator:symbolic-name", "org.eclipse.sensinact.gateway.launcher");
+            configResource.getProperties().put(":configurator:version", "0.0.1");
+        }
+
+        // Remove configuration
+        if (removedConfigurations != null) {
```

Comment:
We should discuss whether this is the right thing to do. Possibly we should only remove them if they haven't been changed by the user? I might get a bit annoyed if I hand edited a config file and the section got automatically deleted.

Reply:
This needs some discussion indeed, as not removing a previous configuration means keeping potentially removed properties